packages/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerCache.java

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.imagepicker;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.net.Uri;
+import androidx.annotation.Nullable;
+import io.flutter.plugin.common.MethodCall;
+import java.util.HashMap;
+import java.util.Map;
+
+class ImagePickerCache {
+
+  static final String MAP_KEY_PATH = "path";
+  static final String MAP_KEY_MAX_WIDTH = "maxWidth";
+  static final String MAP_KEY_MAX_HEIGHT = "maxHeight";
+  private static final String MAP_KEY_TYPE = "type";
+  private static final String MAP_KEY_ERROR_CODE = "errorCode";
+  private static final String MAP_KEY_ERROR_MESSAGE = "errorMessage";
+
+  private static final String FLUTTER_IMAGE_PICKER_IMAGE_PATH_KEY =
+      "flutter_image_picker_image_path";
+  private static final String SHARED_PREFERENCE_ERROR_CODE_KEY = "flutter_image_picker_error_code";
+  private static final String SHARED_PREFERENCE_ERROR_MESSAGE_KEY =
+      "flutter_image_picker_error_message";
+  private static final String SHARED_PREFERENCE_MAX_WIDTH_KEY = "flutter_image_picker_max_width";
+  private static final String SHARED_PREFERENCE_MAX_HEIGHT_KEY = "flutter_image_picker_max_height";
+  private static final String SHARED_PREFERENCE_TYPE_KEY = "flutter_image_picker_type";
+  private static final String SHARED_PREFERENCE_PENDING_IMAGE_URI_PATH_KEY =
+      "flutter_image_picker_pending_image_uri";
+  private static final String SHARED_PREFERENCES_NAME = "flutter_image_picker_shared_preference";
+
+  private static SharedPreferences getFilePref;
+
+  static void setUpWithActivity(Activity activity) {
+    getFilePref =
+        activity
+            .getApplicationContext()
+            .getSharedPreferences(SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE);
+  }
+
+  static void saveTypeWithMehtodCallName(String methodCallName) {
+    if (methodCallName.equals(ImagePickerPlugin.METHOD_CALL_IMAGE)) {
+      setType("image");
+    } else if (methodCallName.equals(ImagePickerPlugin.METHOD_CALL_VIDEO)) {
+      setType("video");
+    }
+  }
+
+  private static void setType(String type) {
+    if (getFilePref == null) {
+      return;
+    }
+    getFilePref.edit().putString(SHARED_PREFERENCE_TYPE_KEY, type).apply();
+  }
+
+  static void saveDemensionWithMethodCall(MethodCall methodCall) {
+    Double maxWidth = methodCall.argument("maxWidth");
+    Double maxHeight = methodCall.argument("maxHeight");
+    setMaxDimension(maxWidth, maxHeight);
+  }
+
+  private static void setMaxDimension(Double maxWidth, Double maxHeight) {
+    if (getFilePref != null) {
```

Comment:
subjective: consider early returns/guard clauses here instead of the extra nesting